src/ship.cpp

```diff
@@ -0,0 +1,52 @@
+#include "ship.h"
+
+Ship::Ship() {
+    this->radius = SHIP_CONSTS::SHIP_RADIUS;
+    this->health = SHIP_CONSTS::INITIAL_HEALTH;
+
+    if (!(this->texture).loadFromFile(SHIP_CONSTS::PATH_TO_TEXTURE, SHIP_CONSTS::TEXTURE_RECT))
```

Comment:
ideally not being able to load a texture should be a game ending condition so I would prefer if you move this out of the constructor and into a seperate init() function and check for ret code for better error handling.